tests/conftest.py

```diff
@@ -1332,7 +1353,9 @@ def recorder_db_url(
         assert not sqlalchemy_utils.database_exists(db_url)
         sqlalchemy_utils.create_database(db_url, encoding="utf8")
     yield db_url
-    if db_url.startswith("mysql://"):
+    if db_url == "sqlite://" and persistent_database:
+        rmtree(tmp_path, ignore_errors=True)
```

Comment:
Its unexpected that tmp_path_factory doesn't clean up the temp dir automatically

Reply:
tmp_path_factory is a session level fixture. I use it here to avoid creating temporary directories which in most case won't be needed.